src/server/game/Entities/Player/Player.cpp

```diff
@@ -16766,6 +16766,8 @@ void Player::SendQuestUpdateAddCreatureOrGo(Quest const* quest, ObjectGuid guid,
     uint16 log_slot = FindQuestSlot(quest->GetQuestId());
     if (log_slot < MAX_QUEST_LOG_SIZE)
         SetQuestSlotCounter(log_slot, creatureOrGO_idx, GetQuestSlotCounter(log_slot, creatureOrGO_idx) + add_count);
+
+    sScriptMgr->OnPlayerReachedQuestObjectiveCount(this, quest->GetQuestId(), creatureOrGO_idx, old_count + add_count);
```

Comment:
The value you are using here, `creatureOrGO_idx`, is not creature/gameobject entry but index of the objective being completed in quest_template

Now, I think it is actually better to use the objective index because it will let you also grab required count from  quest template